src/main/asciidoc/reference/elasticsearch-misc.adoc

```diff
@@ -23,50 +23,49 @@ Page<SampleEntity> sampleEntities =
+Elasticsearch has a scroll API for getting big result set in chunks. `ElasticsearchTemplate` has startScroll and continueScroll methods that can be used as below.
 
-.Using Scan and Scroll
+.Using startScroll and continueScroll
 ====
 [source,java]
 ----
-SearchQuery searchQuery = new NativeSearchQueryBuilder()
-    .withQuery(matchAllQuery())
-    .withIndices("test-index")
-    .withTypes("test-type")
-    .withPageable(new PageRequest(0,1))
-    .build();
-String scrollId = elasticsearchTemplate.scan(searchQuery,1000,false);
-List<SampleEntity> sampleEntities = new ArrayList<SampleEntity>();
-boolean hasRecords = true;
-while (hasRecords){
-    Page<SampleEntity> page = elasticsearchTemplate.scroll(scrollId, 5000L , new ResultsMapper<SampleEntity>()
-    {
-        @Override
-        public Page<SampleEntity> mapResults(SearchResponse response) {
-            List<SampleEntity> chunk = new ArrayList<SampleEntity>();
-            for(SearchHit searchHit : response.getHits()){
-                if(response.getHits().getHits().length <= 0) {
-                    return null;
-                }
-                SampleEntity user = new SampleEntity();
-                user.setId(searchHit.getId());
-                user.setMessage((String)searchHit.getSource().get("message"));
-                chunk.add(user);
-            }
-            return new PageImpl<SampleEntity>(chunk);
-        }
-    });
-    if(page != null) {
-        sampleEntities.addAll(page.getContent());
-        hasRecords = page.hasNextPage();
-    }
-    else{
-        hasRecords = false;
-    }
-    }
-}            
+SearchResultMapper searchResultMapper = new SearchResultMapper() {
+	@Override
+	public <T> AggregatedPage<T> mapResults(SearchResponse response, Class<T> clazz, Pageable pageable) {
+		List<SampleEntity> result = new ArrayList<SampleEntity>();
+		for (SearchHit searchHit : response.getHits()) {
+			if (response.getHits().getHits().length <= 0) {
+				return new AggregatedPageImpl<T>(Collections.EMPTY_LIST, response.getScrollId());
+			}
+			String message = (String) searchHit.getSourceAsMap().get("message");
+			SampleEntity sampleEntity = new SampleEntity();
+			sampleEntity.setId(searchHit.getId());
+			sampleEntity.setMessage(message);
+			result.add(sampleEntity);
+		}
+		if (result.size() > 0) {
+			return new AggregatedPageImpl<T>((List<T>) result, response.getScrollId());
+		}
+		return new AggregatedPageImpl<T>(Collections.EMPTY_LIST, response.getScrollId());
+	}
+};
+SearchQuery searchQuery = new NativeSearchQueryBuilder().withQuery(matchAllQuery())
+			.withIndices(INDEX_NAME)
+			.withTypes(TYPE_NAME)
+			.withFields("message")
+			.withPageable(PageRequest.of(0, 10))
+			.build();
+Page<SampleEntity> scroll = elasticsearchTemplate.startScroll(1000, searchQuery, SampleEntity.class, searchResultMapper);
```

Comment:
Please add a new line before for better readability.

Reply:
Done